app/main/views/suppliers.py

```diff
@@ -22,9 +22,13 @@ def list_suppliers():
     suppliers = Supplier.query.order_by(Supplier.name)
 
     if prefix:
-        # case insensitive LIKE comparison for matching supplier names
-        suppliers = suppliers.filter(
-            Supplier.name.ilike(prefix + '%'))
+        if prefix == 'other':
+            suppliers = suppliers.filter(
+                Supplier.name.op('~')('^[^A-z]+'))
```

Comment:
It looks as though just [A-Z] works case-insensitively with Postgres regexes, but yes it's probably better to be explicit about lowercase:
```
gdmdev=# select name from supplier where name ~* '^[^A-Z]';
            name
-----------------------------
 4sl Group
 100 Percent IT Ltd
 5NINES Data Centres Limited
 3gamma Limited
 4C Strategies Ltd
 1Tech Limited
 365 ITMS Ltd
 3M United Kingdom
 4Projects
 2T Security Ltd
 23 Technology Ltd
 6 POINT 6
 247 Time Ltd
 365 Response
 2nd Quadrant Limited
 3Cubed Limited
 1st Milestone Ltd
 1Spatial Group Ltd
 3R Telecom Limited
 8x8 Solutions
 4Secure
 3Zero Solutions Limited
(22 rows)
```